src/Admin/AjaxAction.php

```diff
@@ -0,0 +1,225 @@
+<?php
+/**
+ * Class AjaxAction.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\Admin;
+
+/**
+ * Base class to define a new AJAX action.
+ *
+ * @package AmpProject\AmpWP
+ */
+final class AjaxAction {
+
+
+	/**
+	 * The AJAX action should be accessible by any visitor.
+	 *
+	 * @var int
+	 */
+	const ACCESS_ANY = 0;
+
+	/**
+	 * The AJAX action should only be accessible by authenticated users.
+	 *
+	 * @var int
+	 */
+	const ACCESS_USER = 1;
+
+	/**
+	 * The AJAX action should only be accessible by unauthenticated visitors.
+	 *
+	 * @var int
+	 */
+	const ACCESS_NO_USER = 2;
```

Comment:
I think we can just restrict this to authenticated users, as we don't have a use case for unauthenticated admin actions. These could be supported later if we need to.